src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -823,7 +823,19 @@ private function syncNode($uuid, $path, $type, $isNewNode, $props = array(), $pr
             $this->syncBinaryData($nodeId, $propsData['binaryData']);
         }
 
-        if (!empty($propsData['references'])) {
+        $needSyncReferences = !empty($propsData['references']);
+        if (!$needSyncReferences) {
+            foreach ($props as $property) {
+                if (in_array($property->getType(), array(PropertyType::REFERENCE, PropertyType::WEAKREFERENCE))) {
+                    // there are no references from this node anymore, which means there is a multivalue
+                    // reference property with an empty array. need to sync to remove previous references
```

Comment:
not sure about the performance penalty of always syncing. alternatively, the node would need to track if it had any reference properties when it was loaded, but that sounds complicated and error prone.